### ISSUE TITLE
Elm app: 'Shutting down tauri process' in 5 seconds after tauri dev

### ISSUE THREAD

**Issue by folex** [tauri-apps/tauri#768]
UPD: Issue solved. Update to latest node (14.5.0) fixed it.
-----

Thank you for an awesome project! Eager to try and use it!

**Describe the bug**
I have an Elm app https://github.com/fluencelabs/p2p-fileshare , and I wanted to try tauri with it. So I run `elm-app start` and then `npx tauri dev` (configured to http:/localhost:3000/`), tauri starts, shows white window, and then stops:

```log
$ npx tauri dev
[tauri]: running dev
(node:90475) ExperimentalWarning: The fs.promises API is experimental
 app:spawn Running "cargo run" +0ms

   Compiling app v0.1.0 (/Users/folex/Development/p2p-fileshare/src-tauri)
   Compiling tauri v0.7.1
    Finished dev [unoptimized + debuginfo] target(s) in 5.91s
     Running `target/debug/app`
 app:tauri Shutting down tauri process... +6s
```

**To Reproduce**
1. Clone [p2p-fileshare](https://github.com/fluencelabs/p2p-fileshare)
2. Run `npm install`
3. Configure tauri with: `{ "distDir": "./build", "devPath": "http://localhost:3000/" }`
4. `npm run start` will run `elm-app start`
5. `npx tauri dev`

I'm not sure if that has any effect, but to produce `./build` directory, run `npm run build`, it will generate release bundle (it takes a while).

**Expected behavior**
Tauri development window appear

**Screenshots**
Just a white window which closes pretty fast.

**Platform and Versions (please complete the following information):**
```
$ npx tauri info
[tauri]: running info

Operating System - Darwin(18.7.0) - darwin/x64

Node.js environment
  Node.js - 10.16.0
  tauri.js - 0.8.1

Rust environment
  rustc - 1.46.0-nightly
  cargo - 1.45.0-nightly
  tauri-bundler - Not installed

Global packages
  NPM - 6.14.4
  yarn - 1.17.3

App directory structure
/.circleci
/.git
/build
/elm-stuff
/node_modules
/public
/src
/src-tauri
/tests

App
  tauri.rs - 0.7.1
  mode - embedded-server
  build-type - bundle
  CSP - default-src blob: data: filesystem: ws: http: https: 'unsafe-eval' 'unsafe-inline'
  Windows - Edge
  distDir - ./build
  devPath - http://localhost:3000/
```

**Stack Trace**
```
$ npx tauri dev
[tauri]: running dev
(node:90475) ExperimentalWarning: The fs.promises API is experimental
 app:spawn Running "cargo run" +0ms

   Compiling app v0.1.0 (/Users/folex/Development/p2p-fileshare/src-tauri)
   Compiling tauri v0.7.1
    Finished dev [unoptimized + debuginfo] target(s) in 5.91s
     Running `target/debug/app`
```

Would be awesome to have some `--verbose` option to enable debug logs, or maybe `RUST_LOG=debug` via `env_logger` :)

**Comment by lucasfernog** [comment]
Will take a look at it tomorrow. Sorry I've been busy with Windows :/

I totally agree with you regarding logging, but we're a small team :( maybe you could start a PR starting it?

**Comment by lucasfernog** [comment]
Tried it with the latest published Tauri and it seems to work fine, can you try again?
If it still doesn't work then it might be your macOS version or the nightly Rust.

**Comment by nothingismagick** [comment]
Regarding logging, there is a new feature request. I highly recommend checking it out and helping figure out the requirements:

https://github.com/tauri-apps/tauri/issues/735

**Comment by folex** [comment]
I've tried on tauri 0.7.5 (btw, awesome updates on cli UX!) and stable rust – same behaviour: white window opens and closes.

I asked a friend to try on ubuntu, and it worked (after a few additional `apt-get`s). 

So I guess it's a macOS that's the problem, or maybe some system-wide packages. I'll try to dig into code to debug this problem when I have free time, but can't promise that yet :) 


Regarding logging: I see that #735 is about in-app logging, I was talking more about debugging tauri itself. If I will have a chance to dig into tauri's code, I'll try to submit a PR, but can't promise.


P.S. here's what my friend had to install apart from packages described in the docs:
- libdbus-1-dev
- libgtk-3-dev
- libsoup2.4

hope that helps!

**Comment by nothingismagick** [comment]
![image](https://user-images.githubusercontent.com/35242872/87300044-71a35e00-c50d-11ea-8c9b-439b463262b2.png)

So, I have no problem getting this to work, but on reviewing I noticed a couple things: `npx` is buggy, or rather the CDN / and potentially NPM's move to the Microsoft empire.  It actually wrongly installed an outdated template file, which pulled in tauri.rs: 0.5.

I saw a few other troubling issues with `npm install`, that had to do with potentially corrupted tarball data. 
<img width="1985" alt="Screenshot 2020-07-13 at 13 19 56" src="https://user-images.githubusercontent.com/35242872/87301532-0c9d3780-c510-11ea-8418-a12b63322626.png">

Technically offtopic, but there are also a number of "unpatchable vulns" in `create-elm-app`, most of which have to do with your friend and mine, `lodash`.

Please note: you have not installed the bundler. You can do so with:
`cargo install tauri-bundler`


I did install tauri with `yarn add tauri` and ran `yarn tauri dev`. npx does work as well, however there are (as I mentioned above and [here](https://github.com/npm/npx/issues/29#issuecomment-626738569)) concerns with npx as a part of your toolchain.

**Comment by folex** [comment]
**TL;DR:** Seems like problem was in node version. Update from `10.16.0` to `14.5.0` fixed it.

-----

I forgot to mention in the previous message, but I have installed `tauri-bundler`, it didn't help.

But yarn did. With `yarn tauri dev` it runs! I gotta fix `Content-Security-Policy` for the networking to work, but otherwise everything seems working now.

I guess the problem was `node` version, because running `yarn add tauri` errored with
```
[2/4] 🚚  Fetching packages...
error tauri@0.9.1: The engine "node" is incompatible with this module. Expected version ">= 10.17.0". Got "10.16.0"
```

Now it works with both yarn and npm :) 

Thanks a lot!

**Comment by lucasfernog** [comment]
Couldn't reproduce on Linux with node v10.16.0. Anyway weird that npm doesn't validate the engine.

**Comment by lucasfernog** [comment]
Closing it since we're not encouraging npx anymore and the nodejs CLI is being rewritten in Rust so this issue isn't relevant anymore.
